modules/gapi/test/common/gapi_core_tests_inl.hpp

```diff
@@ -1442,27 +1216,14 @@ TEST_P(SqrtTest, AccuracyTest)
 
 TEST_P(NormalizeTest, Test)
 {
-    auto param = GetParam();
-
-    compare_f cmpF;
-    MatType type, ddepth;
-    cv::Size sz;
-    double a = 0 , b = 0;
-    int norm_type = 0;
-    bool createOut = 0;
-    cv::GCompileArgs compile_args;
-
-    std::tie(cmpF, type, sz, a, b, norm_type, ddepth, createOut, compile_args) = GetParam();
-    int dtype = CV_MAKETYPE(ddepth, CV_MAT_CN(type));
-
-    initMatsRandN(type, sz, dtype, createOut);
+    initMatrixRandN(type, sz, CV_MAKETYPE(ddepth, CV_MAT_CN(type)), createOutputMatrices);
```

Comment:
Maybe it's worth to introduce initialization function which initializes the output with type which have the same number of channels as input? (I found 3 tests which can make use of that)

Reply:
I don't think this is critical at the moment. also, this would likely break the signature pattern of other initialization function and I'm not sure that's a good thing.

I'll look into this however, I guess there should be a better abstraction (there are other cases with quite complex initialization), ideally, we must be able to initialize everything before we get to the test body (e.g. in constructor). But I didn't find a good-looking API for that apart from replacing the whole `GAPI_TEST_FIXTURE` macro with manual test fixture creation (basically, doing more or less the same as the macro does)

Reply:
by the way, we can actually the following in test fixture definition:
```cpp
GAPI_TEST_FIXTURE(NormalizeTest, [this] (int, cv::Size, int, bool) { initMatrixRandN(type, sz, CV_MAKETYPE(ddepth, CV_MAT_CN(type)), createOutputMatrices); }, ...)
```
this will basically do the same thing as we do in the test body but we now leverage a custom init function. so I think this is probably a way to go for such cases. alas, I think the API now looks terribly gigantic, may be it makes sense to simplify it somehow

I propose to address this separately